### PR TITLE
{doc} Refine authoring commands doc

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

Current doc page is out of date. Here is to adjust the doc with latest azure cli developer tooling.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
doc

File: doc/authoring_command_modules/README.md [hunk 1]
```diff
@@ -28,39 +28,23 @@ For example:
 ```
 git clone https://github.com/Azure/azure-cli.git
 cd azure-cli
-virtualenv env
+python -m venv env
 source env/bin/activate
-python scripts/dev_setup.py
+azdev setup -c
```
Comment (qwordy) [inline doc/authoring_command_modules/README.md:33]:
```suggestion
azdev setup -c azure-cli -r azure-cli-extensions
```

File: doc/authoring_command_modules/README.md [hunk 2]
```diff
@@ -79,6 +63,12 @@ Command modules should have the following structure:
 `-- HISTORY.rst
 ```
 
+You could create these files manually, or [create a module with `azdev cli create`](https://azurecliprod.blob.core.windows.net/videos/04%20-%20AzdevCliCreate.mp4).
+
+You also could consider using [Code-gen tool](https://azurecliprod.blob.core.windows.net/videos/04%20-%20AzdevCliCreate.mp4) to generate code automatically.
```
Comment (qwordy) [inline doc/authoring_command_modules/README.md:68]:
```suggestion
You could also consider using [Code-gen tool](https://azurecliprod.blob.core.windows.net/videos/04%20-%20AzdevCliCreate.mp4) to generate code automatically.
```

File: doc/authoring_command_modules/README.md [hunk 3]
```diff
@@ -79,6 +63,12 @@ Command modules should have the following structure:
...
+
+
+
```
Comment (qwordy) [inline doc/authoring_command_modules/README.md:71]:
```suggestion
```

File: doc/authoring_command_modules/README.md [hunk 4]
```diff
@@ -28,39 +28,23 @@ For example:
...
 If your commands aren't showing with `az`, use `az --debug` to help debug. There could have been an exception
 thrown whilst attempting to load your module.
 
 
 <a name="heading_author_command_mod"></a>Authoring command modules
 ------
-Currently, all command modules should start with `azure-cli-`.  
-When the CLI loads, it search for packages installed that start with that prefix.
-
 The `example_module_template` directory gives an example command module with other useful examples.
```
Comment (fengzhou-msft) [inline doc/authoring_command_modules/README.md:48]: The `example_module_template` directory no longer exists, right? This line can also be removed.

File: doc/authoring_command_modules/README.md [hunk 5]
```diff
@@ -28,39 +28,23 @@ For example:
...
-If installation was successful, you should be able to run `pip list` and see your command module.
-```
-$ pip list
-...
-azure-cli-example (0.0.1, /Users/myuser/Repos/azure-cli-example)
-...
-```
-
-Also, you can run `az` and if your command module contributes any commands, they should appear.
+If your command module contributes any commands, they should appear with `az`.
```
Comment (fengzhou-msft) [inline doc/authoring_command_modules/README.md:41]:
```suggestion
If your command module contributes any commands, they should appear when running `az`.
```
Make it more clear with a verb.

File: doc/authoring_command_modules/README.md [hunk 6]
```diff
@@ -79,6 +63,10 @@ Command modules should have the following structure:
 `-- HISTORY.rst
 ```
 
+You could create these files manually, or [create a module with `azdev cli create`](https://azurecliprod.blob.core.windows.net/videos/04%20-%20AzdevCliCreate.mp4).
+
+You could also consider using [Code-gen tool](https://azurecliprod.blob.core.windows.net/videos/04%20-%20AzdevCliCreate.mp4) to generate code automatically.
```
Comment (fengzhou-msft) [inline doc/authoring_command_modules/README.md:68]: The link is not right.

Shall we move `Code-gen` above `azdev cli create` as it is more recommended now?

Comment (Juliehzl) [reply doc/authoring_command_modules/README.md:68]: good check!

File: doc/authoring_command_modules/README.md (hunk 4 above)
Comment (Juliehzl) [reply doc/authoring_command_modules/README.md:48]: yes

File: doc/authoring_command_modules/README.md (hunk 5 above)
Comment (Juliehzl) [reply doc/authoring_command_modules/README.md]: done

**Review by fengzhou-msft — APPROVED** [review]

File: doc/authoring_command_modules/README.md [hunk 7]
```diff
@@ -20,64 +20,35 @@ The document provides instructions and guidelines on how to author command modul
...
-If installation was successful, you should be able to run `pip list` and see your command module.
-```
-$ pip list
-...
-azure-cli-example (0.0.1, /Users/myuser/Repos/azure-cli-example)
-...
-```
-
-Also, you can run `az` and if your command module contributes any commands, they should appear.
+If your command module contributes any commands, they should appear when running `az`.
```
Comment (qwordy) [inline doc/authoring_command_modules/README.md:41]: What is "they"

Comment (Juliehzl) [reply doc/authoring_command_modules/README.md:41]: commands

File: doc/authoring_command_modules/README.md [hunk 8]
```diff
@@ -20,64 +20,35 @@ The document provides instructions and guidelines on how to author command modul
...
 Clone the repo, enter the repo directory then create your virtual environment.  
 
 For example:
 ```
 git clone https://github.com/Azure/azure-cli.git
-cd azure-cli
-virtualenv env
+git clone https://github.com/Azure/azure-cli-extensions.git
+python -m venv env
 source env/bin/activate
```
Comment (qwordy) [inline doc/authoring_command_modules/README.md:32]: You assume it is Linux environment.

Comment (Juliehzl) [reply doc/authoring_command_modules/README.md:32]: Yes. It is a simple example and for more information about setting up environment, link to https://github.com/Azure/azure-cli-dev-tools#setting-up-your-development-environment for details.

Comment (qwordy) [reply doc/authoring_command_modules/README.md:32]: Can you add a hint "This is script for Linux"?

**Review by qwordy — APPROVED** [review]

**Pull request merged**